src/Serialization/LieTheory.jl

```diff
@@ -112,36 +93,27 @@ end
 
 @register_serialization_type WeylGroup uses_id
 
+type_params(W::WeylGroup) = WeylGroup, root_system(W)
```

Comment:
Stupid question maybe: It seems to me that the first return value of `type_params` is always the argument type. Is this a correct observation?
If yes, wouldn't it be possible to refactor that away into the calling function, so that `type_params` loses the first return value? Imo, this could make the code a bit more readable. If there is a reason for this, I am happy to learn about it

Reply:
maybe there is a way to get a round it, but it was the only way I found I could get the new `save_type_params` to work. See main.jl